src/Microsoft.ML.TorchSharp/NasBert/TextClassificationTrainer.cs

```diff
@@ -0,0 +1,857 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.ML.CommandLine;
+using Microsoft.ML.Data;
+using Microsoft.ML.Internal.Utilities;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.TorchSharp.NasBert.Models;
+using Microsoft.ML.TorchSharp.NasBert.Preprocessing;
+using Microsoft.ML.Trainers;
+using Microsoft.ML.Transforms;
+using TorchSharp;
+
+using static TorchSharp.torch;
+using static TorchSharp.torch.nn;
+using static TorchSharp.TensorExtensionMethods;
+using Microsoft.ML.TorchSharp.Utils;
+using Microsoft.ML.TorchSharp.NasBert.Optimizers;
+using Microsoft.ML;
+using Microsoft.ML.TorchSharp.NasBert;
+using Microsoft.ML.TorchSharp.Extensions;
+using System.IO;
+
+[assembly: LoadableClass(typeof(SentenceClassificationTransformer), null, typeof(SignatureLoadModel),
+    SentenceClassificationTransformer.UserName, SentenceClassificationTransformer.LoaderSignature)]
+
+[assembly: LoadableClass(typeof(IRowMapper), typeof(SentenceClassificationTransformer), null, typeof(SignatureLoadRowMapper),
+    SentenceClassificationTransformer.UserName, SentenceClassificationTransformer.LoaderSignature)]
+
+namespace Microsoft.ML.TorchSharp.NasBert
+{
+    public sealed class TextClassificationTrainer : IEstimator<SentenceClassificationTransformer>
+    {
+        private readonly IHost _host;
+        private readonly Options _options;
+        private SentenceClassificationTransformer _transformer;
+        private const string ModelUrl = "models/NasBert2000000.tsm";
+
+        /// <summary>
+        /// Sentence classification model.
+        /// </summary>
+        public enum Architecture
```

Comment:
>public enum Architecture [](http://example.com/codeflow?start=8&length=24)

Is it possible this enum can be used in the future outside 
TextClassificationTrainer class? If this is any possiblity, I would move it outside TextClassificationTrainer and name it something like BertArchitecture instead.

Reply:
That makes sense. Bert is used for all types of NLP, not just classification. I'll move it and we can discuss this during preview if it comes up again.